### ISSUE TITLE
[BUG] Not possible to use string as input for v1/embeddings

### ISSUE THREAD

**Issue by ro99** [theroyallab/tabbyAPI#245]
### OS

Linux

### GPU Library

CUDA 12.x

### Python version

3.11

### Describe the bug

OpenAI API allow the use of ```string``` as input for [v1/embeddings endpoint](https://platform.openai.com/docs/api-reference/embeddings/create). Currently TabbyAPI only considers the **List** object as input:

```python
class EmbeddingsRequest(BaseModel):
    input: List[str] = Field(
        ..., description="List of input texts to generate embeddings for."
    )
    encoding_format: str = Field(
        "float",
        description="Encoding format for the embeddings. "
        "Can be 'float' or 'base64'.",
    )
    model: Optional[str] = Field(
        None,
        description="Name of the embedding model to use. "
        "If not provided, the default model will be used.",
    )
```

Currently when trying to send a string, the system will return http error 422:

```json
{
	"detail": [
		{
			"type": "list_type",
			"loc": [
				"body",
				"input"
			],
			"msg": "Input should be a valid list",
			"input": "The food was delicious and the waiter..."
		}
	]
}
```

This is an issue because some clients use a string as a first validation step before creating an embedding. For instance, obsidian-copilot client receives http error 422 when sending "Sample text for embedding" as input.

### Reproduction steps

Just send a POST request for OAI endpoint ```v1/embeddings``` using the following:

```json
{
    "input": "The food was delicious and the waiter...",
    "model": "multilingual-e5-large-instruct",
    "encoding_format": "float"
}
```

### Expected behavior

When sending a POST http request for ```v1/embeddings```, besides using an array, it should also be possible to use a string as input. For example:

```json
{
    "input": "The food was delicious and the waiter...",
    "model": "multilingual-e5-large-instruct",
    "encoding_format": "float"
}
```

### Acknowledgements

- [X] I have looked for similar issues before submitting this one.
- [X] I have read the disclaimer, and this issue is related to a code bug. If I have a question, I will use the Discord server.
- [X] I understand that the developers have lives and my issue will be answered when possible.
- [X] I understand the developers of this program are human, and I will ask my questions politely.

**Comment by bdashore3** [comment]
Good catch. Missed this in the OAI docs:

https://platform.openai.com/docs/api-reference/embeddings/create

**Comment by bdashore3** [comment]
Fixed in b94c646210d56539cf0814afd2d99051825cc55c
